api/authorization-api-1_0.md

```diff
@@ -328,77 +329,256 @@ The following is a non-normative example of a simple Decision:
+
+A PEP can ignore this field if it doesn't know how to process it.
+
+The Reason object being a JSON object, implementations MUST follow the following structure; the `reason` context contains the following elements:
+
+`id`:
+: REQUIRED. A string value that specifies the reason within the scope of a particular response. In case the PDP returns several reasons, the `id` MUST uniquely identify each one of them. The `id` MAY also match specific reasons held by the PDP, part of a reasons list or dictionnary for example, and used for audit correlations.
+
+`reason_admin`:
+: OPTIONAL. The reason, which MUST NOT be shared with the end user, but useful for administrative purposes that indicates why the access was granted or denied. The value of this field is a Reason Field object ({{reason-field}}).
+
+`reason_user`:
+: OPTIONAL. The reason, which MAY be shared with the end user that indicates why the access was granted or denied. The value of this field is a Reason Field object ({{reason-field}}). 
 
 ##### Reason Field {#reason-field}
-A Reason Field is a JSON object that has keys and values of type `string`. The following are non-normative examples of Reason Field objects:
+A Reason Field is a JSON object with `String` key and value pairs, which has the following form:
+
+`{ "reason-code": "reason-description"}`
+
+where:
+
+`reason-code`:
+: REQUIRED. A String value representing a code that uniquely identifies the error. Implementations that do not use any error-code mechanism SHOULD default to valid HTTP Error codes (see {{RFC7231}} ).
+
+`reason-description`:
+: REQUIRED. A String holding the human-readable natural language description of the reason the decision was made.
+
+The following are non-normative examples of Reason Field objects:
 
 ~~~ json
 {
-  "en": "location restriction violation"
+  "error-123": "Location restriction violation."
 }
 ~~~
-{: #reason-example title="Example Reason"}
+{: #reason-example1 title="Example Reason Field: access denial."}
 
-##### Reason Object {#reason-object}
-A Reason Object specifies a particular reason. It is a JSON object that has the following fields:
+~~~ json
+{
+  "MGR-OK": "Manager access granted."
+}
+~~~
+{: #reason-example2 title="Example Reason Field: access granted."}
 
-`id`:
-: REQUIRED. A string value that specifies the reason within the scope of a particular response.
+Putting it all together, the following is a non-normative example of a Reason Object comprising Reason Fields:
 
-`reason_admin`:
-: OPTIONAL. The reason, which MUST NOT be shared with the user, but useful for administrative purposes that indicates why the access was denied. The value of this field is a Reason Field object ({{reason-field}}).
+~~~ json
+[
+  {
+    "id": "0",
```

Comment:
This feels a bit verbose to me, although I can see the value of using an array of Reason objects to be able to send back more than one reason. I'm still not sure `id` fits a purpose here. The non-normative value you give in the example seems like it's just an ordinal, which feels to me is duplicative of the array position?  I'm not religious about this, but I feel like if we want to keep `id` we should provide a better example.